src/Microsoft.DotNet.Darc/src/Darc/Operations/GetHealthOperation.cs

```diff
@@ -0,0 +1,291 @@
+                }
+            }
+
+            return passed ? Constants.SuccessCode : Constants.ErrorCode;
+        }
+
+        /// <summary>
+        ///     Create a list of metrics to run.
+        /// </summary>
+        /// <param name="channelsToEvaluate">Channels to evaluate</param>
+        /// <param name="reposToEvaluate">Repositories to evaluate</param>
+        /// <returns>List of Func's that, when evaluated, will produce metrics with output.</returns>
+        private List<Func<Task<HealthMetricWithOutput>>> ComputeMetricsToRun(HashSet<string> channelsToEvaluate,
+            HashSet<string> reposToEvaluate, IEnumerable<Subscription> subscriptions,
+            IEnumerable<DefaultChannel> defaultChannels, IEnumerable<Channel> channels)
+        {
+            var metricsToRun = new List<Func<Task<HealthMetricWithOutput>>>();
+
+            metricsToRun.AddRange(ComputeSubscriptionHealthMetricsToRun(channelsToEvaluate, reposToEvaluate, subscriptions, defaultChannels));
+
+            return metricsToRun;
+        }
+
+        /// <summary>
+        ///     Compute the subscription health metrics to run based on the channels and input repositories
+        /// </summary>
+        /// <param name="channelsToEvaluate">Channels in the initial set.</param>
+        /// <param name="reposToEvaluate">Repositories in the initial set.</param>
+        /// <param name="subscriptions">Subscriptions in the build asset registry.</param>
+        /// <returns>List of subscription health metrics</returns>
+        /// <remarks>
+        ///     Subscription health is based on repo and branch.  We need to find all the combinations that
+        ///     that make sense to evaluate.
+        ///     
+        ///     Since this is a target-of-subscription focused metric, use the set of target repos+branches from
+        ///     <paramref name="subscriptions"/> where the target repo is in <paramref name="reposToEvaluate"/> and
+        ///     the source channel is in <paramref name="channelsToEvaluate"/> or the target branch is in branches.
+        ///     Also add in repos (in <paramref name="reposToEvaluate"/>) who have a default channel association targeting
+        ///     a channel in <paramref name="channelsToEvaluate"/>
+        ///     
+        ///     Note that this will currently miss completely untargeted branches, until those have at least one
+        ///     default channel or subscription. This is a fairly benign limitation.
+        /// </remarks>
+        private List<Func<Task<HealthMetricWithOutput>>> ComputeSubscriptionHealthMetricsToRun(HashSet<string> channelsToEvaluate,
+            HashSet<string> reposToEvaluate, IEnumerable<Subscription> subscriptions, IEnumerable<DefaultChannel> defaultChannels)
+        {
+            IRemoteFactory remoteFactory = new RemoteFactory(_options);
+
+            // Compute the combinations that make sense.
+            var repoBranchCombinations = subscriptions
```

Comment:
Would it make sense to keep this as a hashSet  to avoid the distinct operation further down?

Reply:
Primarily because HashSet doesn't have an AddRange.  But I can use UnionWith to keep it readable.